### PR TITLE
Bulk API

### PR DESCRIPTION
**Pull request by igalarzab**

### CODE REVIEW

File: src/api/routes/user.ts [hunk 1]
```diff
@@ -17,11 +12,12 @@ import { ApiError, apiErrorCodes } from '../ApiError';
 import { DomainValidationError } from '../../domain/model/DomainValidationError';
 import { ResourceNotFoundError } from '../../domain/model/ResourceNotFoundError';
 import { UserNotFoundError } from '../../domain/model/user/UserRepository';
+import { UpdateUserCommand } from '../../presentation/dtos/users';
 
 export default () => {
   const route = new AsyncRouter();
 
-  route.get('/users/:id', isAuthenticated, async (req: AuthenticatedRequest, res: Response) => {
+  route.get('/:id', isAuthenticated, async (req: AuthenticatedRequest, res: Response) => {
```
Comment (StevePole) [inline src/api/routes/user.ts:20]: How does the rest of the path get established?  Or are we actually changing the path?

File: src/presentation/api/RootController.ts [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+      .use('/v1', accessPass())
+      .use('/v1/admin', this.adminController.routes())
+      .use('/v1', auth())
+      .use('/v1', countries())
+      .use('/v1', permissions())
+      .use('/v1', roles())
+      .use('/v1', sharingCode())
+      .use('/v1', test())
+      .use('/v1', testTypes())
+      .use('/v1/users', user())
```
Comment (StevePole) [inline src/presentation/api/RootController.ts:37]: Ah, it's here.  Wonder if we should go the whole hog and switch them all if we're taking this approach?

File: src/presentation/api/RootController.ts [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+      .use(this.notFoundHandling())
+      .use(this.errorHandling());
+  }
+
+  private notFoundHandling() {
+    return (_: Request, __: Response, next: (...things: any[]) => any) => {
```
Comment (StevePole) [inline src/presentation/api/RootController.ts:43]: What are these '_' guys?

File: src/presentation/api/AdminController.ts [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+import { UserTransformer } from '../transformers/users';
+
+@Service()
+export class AdminController {
+  constructor(private bulkCreateUsers: BulkCreateUsers, private userTransformer: UserTransformer) {}
+
+  public routes() {
+    const route = new AsyncRouter();
+
+    route.get('/users', isAuthenticated, async (req: AuthenticatedRequest, res: Response) => {
```
Comment (oliverviljamaa) [inline src/presentation/api/AdminController.ts:19]: Should we have a permission check for a new permission, e.g. `BULK_CREATE_USERS`?

Comment (oliverviljamaa) [reply src/presentation/api/AdminController.ts:19]: Or is it configured elsewhere?

Comment (igalarzab) [reply src/presentation/api/AdminController.ts:19]: Yep, I haven't added the permissions yet

File: src/presentation/api/RootController.ts (hunk 3 above)
Comment (igalarzab) [reply src/presentation/api/RootController.ts:43]: params not used

File: src/presentation/api/RootController.ts (hunk 2 above)
Comment (igalarzab) [reply src/presentation/api/RootController.ts:37]: I didn't want to make the PR bigger, but I'll start moving the other stuff in next PRs

**Pull request merged**